src/wayland/client.cpp

```diff
@@ -43,11 +43,11 @@ void mw::Client::register_client(wl_client* raw, std::shared_ptr<Client> const&
 void mw::Client::unregister_client(wl_client* raw)
 {
     auto const map = client_map.lock();
-    map->erase(remove_if(
-            begin(*map),
-            end(*map),
+    map->erase(std::remove_if(
+            map->begin(),
+            map->end(),
             [&](auto& item){ return item.first == raw; }),
-        end(*map));
+        map->end());
```

Comment:
OK, assuming that `begin()`/`end()` are resolved to the ones from `std` that changes nothing. Does it really fix a TICS violation?

I suspect the violation comes from elsewhere: not "knowing" that `operator*()` returns a reference to the same container. (It isn't clear that switching to `operator->()` should be any better.)

We can do better since C++20...

```c++
    std::erase_if(*map, [&](auto const& item){ return item.first == raw; });
```